File: 908-MiddleOfTheLinkedList/908-MiddleOfTheLinkedList.py
# Last updated: 4/29/2026, 10:27:03 AM
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        node_list = []
        node = head
        while node != None:
            node_list.append(node)
            node = node.next
        
        mid = len(node_list) // 2
        return node_list[mid]

File: 908-MiddleOfTheLinkedList/908-MiddleOfTheLinkedList.js
// Last updated: 4/29/2026, 10:27:07 AM
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
// 2022-11-12
// 3:15s
// O(n)
// O(1)
var middleNode = function(head) {
    let node1 = head;
    let node2 = head;
    while(node1.next) {
        node1 = node1.next;
        if(node1.next) {
            node1 = node1.next;
        }
        node2 = node2.next;
    }
    return node2;
};